lib/orphans.nix

```
@@ -0,0 +1,103 @@
+{ lib }:
+let
+  inherit (lib)
+    types
+    optional
+    isDerivation
+    attrByPath
+    flatten
+    any
+    assertMsg
+  ;
+  inherit (builtins)
+    length
+    concatStringsSep
+    mapAttrs
+    attrValues
+    tryEval
+  ;
+  isAttr = types.attrs.check;
+in
+rec {
+  /* Return a list of maintainers from a package, empty list if inacessible
+
+    Example:
+    listMaintainers pkgs.foo
+    => [ { ... } ]
+    listMaintainers pkgs.bar
+    => [ ]
+    listMaintainers {}
+    => error: listMaintainers must receive a package
+  */
+  listMaintainers = p:
+  if isDerivation p then
+    (attrByPath [ "meta" "maintainers" ] [ ] p)
+  else
+    throw "listMaintainers must receive a package"
+  ;
+
+  /* Returns the amount of maintainers that a package have
+
+    Example:
+    maintainersLength pkgs.foo
+    => 1
+    maintainersLength pkgs.bar
+    => 0
+    maintainersLength {}
+    => error: listMaintainers must receive a package
+  */
+  maintainersLength = p:
+    length (listMaintainers p);
+
+  /* Returns if a package has a maintainer (not orphan)
+
+    Example:
+    hasMaintainer pkgs.foo
+    => true
+    hasMaintainer pkgs.bar
+    => false
+    hasMaintainer {}
+    => error: listMaintainers must receive a package
+  */
+  hasMaintainer = p:
+    maintainersLength p > 0;
+
+  /* Returns a list of addresses from the node attr/package of orphan packages
```

```suggestion
  /* Returns a list of addresses from the node attrset/package of orphan packages.
```

A;so, what you mean by addresses?

---

Attset addresses, like how would you access a value on a nest of attrsets. 

I could use attrByPath compatible paths but these are flattened out by the flatten function so if you want it in these form you should split it again and because it's a extra step I omitted. BTW this way it's more useful if you are doing exploratory analysis cause lazy evaluation stuff.